src/mmw/js/src/draw/views.js

```diff
@@ -63,8 +64,14 @@ function validateRwdShape(result) {
 function validateShape(polygon) {
     var area = coreUtils.changeOfAreaUnits(turfArea(polygon), 'm<sup>2</sup>', 'km<sup>2</sup>'),
         d = new $.Deferred();
-
-    if (area > MAX_AREA) {
+    var selfIntersectingShape = turfKinks(polygon).features.length > 0
+
+    if (selfIntersectingShape) {
+        var errorMsg = 'This watershed shape is invalid because it intersects ' +
+            'itself. Try drawing the shape again without crossing over its own ' + 'border.'
```

Comment:
Consider formatting this so the message is left aligned, as done for `var message = 'Sorry, your Area of Interest is too large.'` below.